mypy_django_plugin/lib/helpers.py

```diff
@@ -385,3 +386,36 @@ def add_new_manager_base(api: SemanticAnalyzerPluginInterface, fullname: str) ->
     if sym is not None and isinstance(sym.node, TypeInfo):
         bases = get_django_metadata_bases(sym.node, "manager_bases")
         bases[fullname] = 1
+
+
+def resolve_lazy_reference(
+    reference: str, *, api: Union[TypeChecker, SemanticAnalyzer], django_context: "DjangoContext", ctx: Context
+) -> Optional[TypeInfo]:
+    """
+    Attempts to resolve a lazy reference(e.g. "<app_label>.<object_name>") to a
+    'TypeInfo' instance.
+    """
+    if "." not in reference:
+        # <object_name> -- needs prefix of <app_label>. We can't implicitly solve
+        # what app label this should be, yet.
+        api.fail("Could not resolve lazy reference without an app label", ctx)
```

Comment:
Question: do we need to turn this into an error? Maybe just return `Any`? What are the pros / cons / risks?

Reply:
Base thing is that Django has support without `app_label`, in some cases. But it's not trivial to align with how Django figures out any "current" app label without touching the runtime quite a bit, of course.

Only reason is that I wanted to put a strain on the usage here as we don't always fully align with Django's support. So it's not an error from Django when omitting app label, but we we'll resolve the model as `Type[Any]`. Thing was that from what I could tell Django _always_ support the "full" lazy reference, the one including app label, where the partial one is supported. So intention was to push usage towards using a full one instead, but I guess it could be considered a bit intrusive.

Anyways, I'm not feeling too strong about erroring either so I'm totally fine with having partial support and returning `Type[Any]` without a message when we can't resolve.

Reply:
I think it is better to start without an error. Later we can add them if needed.

Reply:
I've removed the error now